db/db.go

```diff
@@ -11,13 +11,43 @@ type Database interface {
 
 	ListTables(ctx context.Context, schema string) ([]string, error)
 	ListColumns(ctx context.Context, schema, table string) ([]TableColumn, error)
+	ListConstraints(ctx context.Context, schema, table string) ([]TableConstraint, error)
 }
 
 type TableColumn struct {
 	Name     string
 	Type     string
 	Nullable bool
 	Default  string
+	PK       bool
+	FK       string
+	FK_ref   string
+	UK       bool
```

Comment:
Sorry not to have replied sooner. I have little time to contribute for this project (and little free time in general).

You're right, I would normally have created two issues to discuss and 2 PR for this contrib, I just wanted to give some views what could be done to my mind and prototype it. I didn't give more "context" to my contrib.

The story:

My main goal was to prototype constraints on tables such as primary key and foreign keys. Then I just remembered of what schemaspy did : a dot generated ERD. Then I thought about contextualized diagrams with mermaid. And at that point, as I already prototyped it, I decided to submit my work as *DRAFT*, not to be integrated into main code but to give some ideas (you also had them, so that's fine).

Maybe textual information is fine just like PostgreSQL psql   `\d` but having (markdown) tables  with PK and FK seems more visual to me.

Refering tables+column could be displayed in psql way.

Moreover, my main goal would have been to show constraints on columns : references, not nul,  checks and so on.

Reply:
Depending on my free time, I could review this work to be more consistent. But if you want to go faster, feel free to get inspiration, discard my PR or reuse parts (and I'm totally fine with it). If you prefer let me continue this work, you have to know about my little time availability and it may require me weeks to finish (sorry about that).

Reply:
No need to apologize — I'm glad that someone else resonated with the usefulness with this project! I'm probably going to spend some time to make some progress on the project in a few weeks time, but I will surely let you know if I decide to tackle these features specifically.

Thank you!